addons/fleet/models/fleet_vehicle_cost.py

```diff
@@ -128,7 +132,21 @@ def default_get(self, default_fields):
     # (1) to address fields from inherited table
     # (2) fields that aren't stored in database
     cost_amount = fields.Float(related='cost_id.amount', string='Amount', store=True)
-    odometer = fields.Float(string='Odometer at creation', help='Odometer measure of the vehicle at the moment of the contract creation')
+    odometer = fields.Float(string='Odometer at creation', 
+        help='Odometer measure of the vehicle at the moment of the contract creation')
+
+    @api.model
+    def update_contract_state(self):
+        all_contracts = self.search([
+            ('state', 'in', ['open', 'futur', 'toclose']), ('active', '=', True)])
+        today = fields.Date.today()
+        for contract in all_contracts:
+            if contract.start_date and contract.start_date > today:
+                contract.state = 'futur'
+            elif contract.expiration_date and contract.expiration_date < today:
+                contract.state = 'toclose'
+            else:
+                contract.state = 'open'
 
```

Comment:
3 search
futur_contracts = self.search(('state', 'in', ['open', 'toclose'])+date cond).write({'state' : 'futur'})
obj={'1':1}
list=[1]

d ={'a':1}

d=dict()
d['a']=1
d=f(1,2,3,option=1)